### PR TITLE
Fixing a bug in task controller addTask function

### PR DESCRIPTION
**Pull request by KhaledToha**

### CODE REVIEW

**Review by Mohammed-Sbeata — APPROVED** [review]
PRAAAAVOOOO

File: src/controllers/task.ts [hunk 1]
```diff
@@ -28,15 +30,29 @@ const addTaskController = (req: Request, res: Response, next: NextFunction) => {
...
-        data: [
-          taskData,
-        ],
-      })
+      taskData = data.rows[0] as TaskInterface;
+      return Promise.resolve(taskData);
+    })
+    .then(() => checkForMemberInProject(userId, projectId))
+    .then((data: QueryResult) => {
+      if (data.rows.length === 0) {
```
Comment (nada-abuzaid) [inline src/controllers/task.ts:38]:
```suggestion
      if (!data.rows.length) {
```

File: src/controllers/task.ts [hunk 2]
```diff
@@ -7,11 +7,13 @@ import {
   editTaskQuery,
   deleteTaskByIdQuery,
 } from '../database';
-import { TokenRequest, TaskInterface } from '../interfaces';
+import { TokenRequest, TaskInterface, joiInterface } from '../interfaces';
 import { taskSchema, CustomError } from '../helpers';
+import { addProjectUserQuery, checkForMemberInProject } from '../database/query';
 
 const addTaskController = (req: Request, res: Response, next: NextFunction) => {
   const projectId = Number(req.params.id);
```
Comment (nada-abuzaid) [inline src/controllers/task.ts:15]: what about validation here? if the req.params.id = "3N" for example?

**Review by nada-abuzaid — CHANGES_REQUESTED** [review]

File: src/controllers/task.ts [hunk 3]
```diff
@@ -28,15 +30,29 @@ const addTaskController = (req: Request, res: Response, next: NextFunction) => {
...
-      res.status(201).json({
-        message: 'Task Created Successfully',
-        data: [
-          taskData,
-        ],
-      })
+      taskData = data.rows[0] as TaskInterface;
+      return Promise.resolve(taskData);
+    })
+    .then(() => checkForMemberInProject(userId, projectId))
```
Comment (nada-abuzaid) [inline src/controllers/task.ts:36]:
```suggestion
    .then(() => checkForMemberInProject({userId, projectId}))

```

**Review by nada-abuzaid — CHANGES_REQUESTED** [review]

**Review by nada-abuzaid — APPROVED** [review]
شكرا يا خالد

**Pull request merged**